### DISCUSSION TITLE
Yoto Alarms

### DISCUSSION THREAD

**Discussion by harrolf** [cdnninja/yoto_ha#116] [Ideas]
First of all, Thank you and I love your effort in the Yoto integration!

As an enhancement I would like to see *alarms*. Visible, editable and a toggle to disable/enable.

Thanks!

**Comment by cdnninja** [comment]
That should be possible! I'll draft something up in the coming weeks.

**Comment by cdnninja** [comment]
How would you want this displayed?  Or would a service call and attributes on a sensor work for the data? 

No type exist that makes this easy.

**Comment by harrolf** [comment]
Ideally it would look like how apple shows an overview of the alarms and how you can disable/enable them. I didn't think about how this can be managed in automations or UI in HA

For a start a generic enable disable/enable alarm(s) would suffice, this can be easily integrated in automations 👍

**Comment by cdnninja** [comment]
I have been slow to get to this.  I haven't forgotten.  I have started drafting the code to collect this data here: https://github.com/cdnninja/yoto_api/pull/94

Once the data for existing alarms is working and collecting it I'll play with trying to adjust them.

**Reply by harrolf** [reply]
Thank you!

**Comment by cdnninja** [comment]
Master branch now has switches for alarms.    

The toggle doesn't action the change yet. Just displays.  I'll add the on off portion soon.  

Feel free to update to master to test out display aspect.

**Comment by cdnninja** [comment]
I'm also interested in feedback on how they are displayed and if other data is needed.   Right now it's just alarm 1,2,3 etc.  I suspect more data will be needed.

**Comment by cdnninja** [comment]
Toggle now functions.   Let me know what else would be handy on alarms and I can look at adding.  All the core data is now built out so straightforward to add more. Just need to know how to display it that makes sense for your use case.

**Reply by connormxy** [reply]
I think that would be helpful for alarms is to be able to at least keep track of the time that the alarm is set for automatically in the integration or ideally to be able to track and set the time of the alarm. An input much like the time entry input for the start and end of night mode and daytime mode for instance would be helpful there. At the risk of adding more entities, I’m not sure if there’s better way to template this out separately maybe he has a card that I would embed both the time and the toggle. But adding for each alarm, an individual time input to determine the time that the alarm would fire. Currently a toggle for using an alarm appears but one has to remember what the meaning of alarm one, alarm two, alarm three etc. means for the days of the week and the time entry. it does start to become future rich and almost feels like it needs to become its own device, or if you can actually designate a segment of the entity groups within the overall device page 2 alarms separately it might be helpful instead of adding potentially a dozen additional entities on top of all the other controls. I’m not sure if that’s configurable by the developer. But for identity to be able to select the days of the week using a checkmark style and to at least input the time a.m. and p.m. that the alarm should fire and somehow identify that as the same alarm with a toggle, having those three entities for each alarm would be helpful instead of just the toggle itself. So I don’t have to go into the app to adjust the actual time of the alarm or the days of the week of the alarm. Unclear, if that’s in the MqTT browser if I were to go looking and maybe template that myself.

**Comment by cdnninja** [comment]
@harrolf  Any feedback on alarm functionality or if you would like other parts of it added?

**Reply by davida72** [reply]
I'd love to see an alarm function. It's sorely missing in the native functionality. 

I've created an automation to serve as an alarm but it would be neater to have it in the integration I think.

**Reply by cdnninja** [reply]
On off exists today for alarms.  Are you looking to edit time of alarms or create new?

**Reply by davida72** [reply]
I'd like to set my Yoto to play from a card at a set time without an automation. 

I posted my automation in the Discussions forum.

**Reply by cdnninja** [reply]
Hmm. Without an automation? So natively in Yoto?  Maybe I'm lost but Yoto feature requests would need to go to them.  Automations just use Yotos native functions to do more.  

I did see the bug on the one play card issue I'll look at though.

**Reply by davida72** [reply]
Yes, I was hoping for some native functionality in the Yoto itself.

**Reply by cdnninja** [reply]
You will need to reach out to yoto.

**Reply by davida72** [reply]
Ok. Thx.
